internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -798,6 +802,8 @@ func (r *AgentServiceConfigReconciler) newImageServiceDeployment(ctx context.Con
 		deployment.Spec.Template.Spec.Containers = []corev1.Container{container}
 		deployment.Spec.Template.Spec.Volumes = volumes
 		deployment.Spec.Template.Spec.ServiceAccountName = imageServiceName
+		deployment.Spec.Template.Spec.NodeSelector = r.NodeSelector
```

Comment:
I suggest making copies. The resource will end up in the client cache, and who knows what may happen to it from there. We don't want some future changes to leak into the map and slice that are stored on the Reconciler.